consensus_engine/consensus.proto

```diff
@@ -110,10 +135,29 @@ message FinalizeResponse {
 
 service ConsensusEngine {
     rpc GetAuthor(GetAuthorRequest) returns(GetAuthorResponse);
-    rpc VerifyHeader(stream VerifyHeaderRequest) returns(stream VerifyHeaderResponse);
-    rpc VerifyHeaders(stream VerifyHeadersRequest) returns(stream VerifyHeadersResponse);
+
+    rpc ChainSpec(google.protobuf.Empty) returns(ChainSpecMessage);
+
+    // Core requests verifications from the Consensus Engine via this function
+    rpc VerifyHeaders(stream VerifyHeaderRequest) returns(stream VerifyHeaderResponse);
+
+    // Consensis Engine may ask for extra informaton (more headers) from the core, and these requests are coming through the stream
+    // returned by the ProvideHeaders function
+    rpc ProvideHeaders(stream HeadersResponse) returns(stream HeadersRequest);
```

Comment:
Shouldn't `HeadersResponse` and `HeadersRequest` switched here?

Reply:
It is weird, but no. I will rework this anyway, but the idea is that the Core initiates the bi-directional stream, where Consensus Engine sends header requests and Core answers with header responses.